pkg/driver/helvarnet/health_test.go

```diff
@@ -0,0 +1,541 @@
+package helvarnet
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smart-core-os/sc-bos/internal/manage/devices"
+	"github.com/smart-core-os/sc-bos/pkg/gen"
+	"github.com/smart-core-os/sc-bos/pkg/gentrait/devicespb"
+	"github.com/smart-core-os/sc-bos/pkg/gentrait/healthpb"
+	"github.com/smart-core-os/sc-golang/pkg/masks"
+	"github.com/smart-core-os/sc-golang/pkg/resource"
+	"github.com/smart-core-os/sc-golang/pkg/wrap"
+)
+
+func setup(devs *devicespb.Collection) *healthpb.Registry {
+
+	return healthpb.NewRegistry(
+		healthpb.WithOnCheckCreate(func(name string, c *gen.HealthCheck) *gen.HealthCheck {
+			_, _ = devs.Update(&gen.Device{Name: name}, resource.WithMerger(func(mask *masks.FieldUpdater, dst, src proto.Message) {
+				dstDev := dst.(*gen.Device)
+				dstDev.HealthChecks = healthpb.MergeChecks(mask.Merge, dstDev.HealthChecks, c)
+			}), resource.WithCreateIfAbsent(), resource.WithExpectAbsent())
+			return nil
+		}),
+		healthpb.WithOnCheckUpdate(func(name string, c *gen.HealthCheck) {
+			_, _ = devs.Update(&gen.Device{Name: name}, resource.WithMerger(func(mask *masks.FieldUpdater, dst, src proto.Message) {
+				dstDev := dst.(*gen.Device)
+				dstDev.HealthChecks = healthpb.MergeChecks(mask.Merge, dstDev.HealthChecks, c)
+			}))
+		}),
+		healthpb.WithOnCheckDelete(func(name, id string) {
+			_, _ = devs.Update(&gen.Device{Name: name}, resource.WithMerger(func(mask *masks.FieldUpdater, dst, src proto.Message) {
+				dstDev := dst.(*gen.Device)
+				dstDev.HealthChecks = healthpb.RemoveCheck(dstDev.HealthChecks, id)
+			}), resource.WithAllowMissing(true))
+		}),
+	)
+}
+
+func TestDeviceNoError(t *testing.T) {
+
+	devs := devicespb.NewCollection()
+	server := devices.NewServer(devicesServerModel{Collection: devs})
+	deviceName := "helvarnet-device-1"
+	reg := setup(devs)
+	exampleChecks := reg.ForOwner("example")
```

Comment:
`exampleChecks` is a strange choice for this var name, maybe something with test instead?